### PR TITLE
Separate page for patient consent records

### LINKED ISSUE #7812: Enhancement to the Existing consent form Implementation
_The issue was created after a call with @aparnacoronasafe understanding user difficulties and requirement_

**Is your feature request related to a problem? Please describe.**
Ref: #7809 , #7416 
Currently, the consent form is part of the edit consultation form, which is not user-friendly and reported following problems such as:
- since it is not shown during the create consultation, the user assumes that, there is no function to upload patient consent in the platform
- The second concern is that, as it is present only in the edit consultation, users are not using the function and they are not alerted that the consent is missing in the consultation

**Describe the solution you'd like**

- [ ] Add a new button called "Consent Records" in the Manage patient dropdown below edit consultation (refer: SS 1)
- [ ] Create a dedicated page called "Consent Record"
- [ ] Replace the Consent form section from the edit consultation form to the new Manage Consents page.
- [ ] Remove the consent form preview from the overview section of the consultation dashboard
- [ ] In the file upload component of the consent form, add an option to select consent type and patient code (applicable only for patient code consent type) Refer: SS 3
- [ ] Add consent type and patient code status(If available) details along with the existing file details such as File name, created and created (Refer: SS 4)
- [ ] When consent is missing for a live admitted patient, show a consent missing badge similar to the review missed badge in the patient consultation page (Refer: SS 2)

Note:
- It will have similar behavior to the existing file upload ( Access controls, functionality, and all )
- It is an optional field and will apply to all patients (For now)
- Currently, Each patient consent type has individual archives and unarchive sections, we will be combining them into one view like our file upload section
- file upload dropdown for patient consent file selection **UI needs to be discussed on call and get approved before starting working on it.**

**Describe alternatives you've considered**

image
image
image
image

### PR DESCRIPTION
**Pull request by shivankacker**
## Proposed Changes

- Fixes #7812

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [x] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA

### CODE REVIEW

File: src/Components/Patient/PatientInfoCard.tsx [hunk 1]
```diff
@@ -634,6 +646,12 @@ export default function PatientInfoCard(props: {
                       consultation?.id &&
                       !consultation?.discharge_date,
                   ],
+                  [
+                    `/facility/${patient.facility}/patient/${patient.id}/consultation/${consultation?.id}/consent-records`,
+                    "Edit Consent Records",
```
Comment (nihal467) [inline src/Components/Patient/PatientInfoCard.tsx]:
```suggestion
                    "Consent Records",
```

as per issue

**Comment by aparnacoronasafe** [comment]
Issues noted:

1- when "patient code status for a patient is updated for the first time (within an encounter), this pop-up window need not come up
image

2- when I type file name before selecting file, the name of the file I select replaces the file name I typed. This shouldn't happen. Only when the file name field is empty should the name of file uploaded come up there

3- when I uploaded patient code for the first time "permission denied" error came up alongside success message. The first workflow was good, but the error message is misplaced 

4- small alignment issue view button on mobile screen
image

5- I first filed a patient code status DNH. Then I filed one for active treatment. The heading - patient code status continued to show DNH. When I tried to update the patient code status the pop up window also said I am changing it from DNH (it should have been active treatment since that is the latest code status)
image

6- each document uploaded for patient code status must say what kind of code status it was. Most likely the nurses won't be writing the detail in file name. Then as the user looks at the log of file for code status, we will have no clue what kind they are! 

7- when user deletes a consent record, we need a greyed text appear in the log " name of user (username) deleted a record." 
If it's code status, mention " name of user (username) deleted a record for <type of code status> code status"

**Comment by khavinshankar** [comment]
@skks1212 can you make the above recommended changes

cc: @nihal467

**Comment by shivankacker** [comment]
@aparnacoronasafe changes have been made.
I have replaced "delete" with archive. This will allow you to see archived consents in the new tab.

I could not add the "nurse changed x" logs as that would require a backend PR. If the current solution does not work, then I will make that.

**Comment by nihal467** [comment]
image

- [ ] keep the archive individual file level
- [ ] create a backend PR to record who uploaded and arcive the respective file, ( Front-end changes to reflect this data can be done next week as enhancement )

**Comment by shivankacker** [comment]
@nihal467 changes made

**Comment by nihal467** [comment]
image

- [ ] keep only individual file level archive only

image

- [ ] can we keep the file preview ( Left side ) and file upload component (Right side) into same level , expected behavior screenshot is attached in the bottom

image

@skks1212

**Review by nihal467 — CHANGES_REQUESTED** [review]
make the minor changes in the comment

**Comment by nihal467** [comment]
@skks1212 
image

when i click on the consent record, the bed pop-up is triggered, it is not a necessary to have a bed assigned to the patient

image

when i archive a file, the page state is not changing, i need to manually refresh the page to view the archive file changes

**Review by rithviknishad — APPROVED** [review]
LGTM

**Comment by nihal467** [comment]
image



- [ ] once the file is archived, make the button similar to file upload

image

- [ ] any file uploaded by a user, should be only allowed to archive by the same user, and admin-only users ( Similar behavior is used in the file upload component)

image

- [ ] if there is no file available, it should be showing a white card saying no file available similar behavior in the file upload component)

image

**Comment by khavinshankar** [comment]
- [x] No records found
image
- [x] Hide Archive button for non-admin users (except for the original user who uploaded it)
image

**Comment by nihal467** [comment]
Approved due to priority, enhancement issue will be opened after merging

**Review by nihal467 — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**